packages/passport/sdk/src/Passport.ts

```diff
@@ -138,14 +139,29 @@ export class Passport {
   }
 
   public connectEvm(): Provider {
-    return new ZkEvmProvider({
+    const provider = new ZkEvmProvider({
       passportEventEmitter: this.passportEventEmitter,
       authManager: this.authManager,
       magicAdapter: this.magicAdapter,
       config: this.config,
       multiRollupApiClients: this.multiRollupApiClients,
       guardianClient: this.guardianClient,
     });
+
+    // EIP-6963 Multi Injected Provider Discovery
+    if (this.config.announceProvider) {
+      announceProvider({
```

Comment:
The `Reference Implementation` in the [EIP-6963 docs](https://eips.ethereum.org/EIPS/eip-6963#:~:text=window.ethereum%20namespace.-,Reference%20Implementation,-Wallet%20Provider) suggest that the provider should be announced on page load, but this implementation only announces the provider after `connectEvm` has been called. 

This seems to go against the intended use case of the spec, and if we advertise Passport as EIP-6963 compliant, then I worry that consumers may not know that `connectEvm` has to be called for the provider to be announced

Reply:
`announceProvider` includes a reference to the `provider` itself so we can't announce before instantiating the provider itself. Some customers may not want zkEVM but instead want an IMXProvider so I think we don't want to announce and create an instance unnecessarily.
  
My reading of EIP-6963 was that it solves the problem of race conditions so the providers can be announced at any stage as they respond to a requestProviders call. I agree we ideally want to fire this asap, but didn't want to impact the flow of calling `connectEVM`.

Reply:
I agree with the potential disconnect on when the `announceProvider` fires based on the config property.. something we could document well? Not sure of a better way here work with the provider instance output from `connectEVM`?

Reply:
@haydenfowler Something that is relatively unique here (maybe not, but haven't seen other solutions like this in the wild) is that our solution is SDK based. There is just more room for variation in how the SDK is bundled/loaded, instantiated and the provider created. Majority of injected providers will be coming from browser extensions that can hook into that page load event.. I feel like we could get away with making sure it's documented, or another option could be putting the `announceProvider` flag inside the `connectEvm` function. Making the flag more contextual to the creation of the provider?